provision/kubernetes/deploy.go

```diff
@@ -433,17 +436,88 @@ func validateHC(hc *provTypes.TsuruYamlHealthcheck, client *ClusterClient) error
 		if hc.TimeoutSeconds >= hc.IntervalSeconds {
 			hc.IntervalSeconds = hc.TimeoutSeconds + 1
 		}
+
+		if err := ensureBackendConfig(ctx, client, app, hc); err != nil {
```

Comment:
In my opinion this call shouldn't be here. A good place for it might be right before ensuring services exist in: https://github.com/tsuru/tsuru/blob/6dd56808976bd8ee00dd35eac5a26f4dea288405/provision/kubernetes/deploy.go#L1243

It's okay to call `version.TsuruYamlData()` again if needed.